app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,74 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
```

Comment:
- [ ] it uses `before_action` to restrict access to the `:new` and `:create` methods
![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png)